job-runner/chain/traverser.go

```diff
@@ -377,3 +386,36 @@ func (t *traverser) sendJL(pJob proto.Job, err error) {
 		jLogger.Errorf("problem sending job log (%#v) to the RM: %s", jl, err)
 	}
 }
+
+func (t *traverser) retrySequence(j proto.Job) {
+	sequenceStartJob := t.chain.SequenceStartJob(j)
+
+	// Create list of jobs in sequence that have been ran and we need to undo
+	var jobsToUndo []string
+	jobsToUndo = append(jobsToUndo, j.Id)
+	currSeqJobId := sequenceStartJob.Id
+	for currSeqJobId != j.Id {
+		jobsToUndo = append(jobsToUndo, currSeqJobId)
+		nextJobs := t.chain.NextJobs(currSeqJobId)
+		if len(nextJobs) != 1 {
+			// error
+		}
+		currSeqJobId = nextJobs[0].Id
```

Comment:
Intention here (and this code block in general) is not clear to me. We're setting the current sequence job ID to the next job's ID... I guess I don't know what the purpose/intention of a "sequence job"?

Reply:
it's a job in the sequence. i've reworked the loop to make it more clear.